src/components/PromotedActionsBar.tsx

```diff
@@ -0,0 +1,147 @@
+import React, {useState} from 'react';
+import type {StyleProp, ViewStyle} from 'react-native';
+import {View} from 'react-native';
+import useLocalize from '@hooks/useLocalize';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as HeaderUtils from '@libs/HeaderUtils';
+import * as ReportActions from '@userActions/Report';
+import type OnyxReport from '@src/types/onyx/Report';
+import Button from './Button';
+import ConfirmModal from './ConfirmModal';
+import type {ThreeDotsMenuItem} from './HeaderWithBackButton/types';
+import * as Expensicons from './Icon/Expensicons';
+
+type PromotedAction = {
+    key: string;
+} & ThreeDotsMenuItem;
+
+type ReportPromotedAction = (report: OnyxReport) => PromotedAction;
+
+type PromotedActionsType = {
+    pin: ReportPromotedAction;
+    // message: (accountID: number) => PromotedAction;
+    // join: ReportPromotedAction;
+    // share: ReportPromotedAction;
+    // hold: () => PromotedAction;
+};
+
+const PromotedActions = {
+    pin: (report) => ({
+        key: 'pin',
+        ...HeaderUtils.getPinMenuItem(report),
+    }),
+    // TODO: Uncomment the following lines when the corresponding features are implemented
+    // message: (accountID) => ({
+    //     key: 'message',
+    //     icon: Expensicons.CommentBubbles,
+    //     text: Localize.translateLocal('common.message'),
+    //     onSelected: () => ReportActions.navigateToAndOpenReportWithAccountIDs([accountID]),
```

Comment:
it's best practice to not leave TODOs, will it be too much to coordinate to add these in the subsequent PRs instead?

Reply:
I'll remove the TODOs. Do you think I should also remove the icons that we are not using in this PR? Shawn provided those, so I included them in this PR.